### PR TITLE
Added campusconcourse scraper for sites with pdf/doc files

### PR DESCRIPTION
**Pull request by HarrisonGregg**
At this point, this is just for [Northampton](https://northampton.campusconcourse.com/search), which has download buttons for doc/pdf files instead of links to HTML pages.

### CODE REVIEW

File: osp_scraper/spiders/campusconcourse_with_files.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+
+import scrapy
+
+from ..spiders.CustomSpider import CustomSpider
+
+class CampusConcourseWithFilesSpider(CustomSpider):
+    name = 'campusconcourse_with_files'
+
+    def start_requests(self):
```
Comment (jodizzle) [inline osp_scraper/spiders/campusconcourse_with_files.py]: Do you mean to have both a `start_requests` and a `parse` in the same file?

File: osp_scraper/spiders/campusconcourse_with_files.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                method='GET',
+                formdata={
+                    'search_performed': '1'
+                }
+            )
+
+    def parse(self, response):
+        pages_text = response.css('.panel-footer label::text').extract_first()
+        num_pages = int(pages_text.split()[-1])
+        print(num_pages)
```
Comment (jodizzle) [inline osp_scraper/spiders/campusconcourse_with_files.py]: Test code?

File: osp_scraper/spiders/campusconcourse_with_files.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                method='GET',
+                formdata={
+                    'search_performed': '1',
+                    'offset': str(page)
+                },
+                meta={
+                    'source_url': response.url,
+                    'source_anchor': 'Search Page ' + str(page),
+                    'depth': 0,
+                    'hops_from_seed': 1,
```
Comment (jodizzle) [inline osp_scraper/spiders/campusconcourse_with_files.py]: Set `depth` and `hops_from_seed` to be the same.

**Comment by HarrisonGregg** [comment]
@jodizzle I think that it does work, scrapy just prioritizes the requests strangely.  It seems like it doesn't follow the redirects immediately, so it queues up a huge number of redirected requests before making the requests and saving the WARC files.  You can see it save them faster, if you limit the number of pages or files downloaded per page, but otherwise it will save them all eventually.

I made the other changes you requested though.  How does it look now?

**Comment by jodizzle** [comment]
@HarrisonGregg I see, I've got it working now.  Looks ready to merge.

**Pull request merged**